### PR TITLE
[QoL] One Pokerus Pokemon for Every Generation

### PR DESCRIPTION
**Pull request by mKroogz**
## What are the changes the user will see?
Pokerus is spread out 1 pokemon per gen on pokerus generation.

## Why am I making these changes?
This is a quite highly requested feature and I believe in the reasoning behind it.
-Being able to have a pokerus mon for single gen challenges
-Encouraging people to use more mons spread across gens
-WE NO LONGER HAVE TO CHECK FOR DUPES IN GENERATION LOGIC
-Honestly doesn't break balance especially still randomized
-Again just giga requested :)

## What are the changes from a developer perspective?
-Changed Pokerus cursor objects to 9 instead of 3 (1 per gen)
-Changed generation loop and logic to be gen based and use "this.allspecies" for filtered by gen mons
-NO LONGER DUPE CHECK 😎  (because each of the 9 mons will be from a specific gen so no dupe chance)

### Screenshots/Videos
image
image
image

## How to test the changes?
Basically just boot it up!  
You can see all 9 pokemon whether you've caught them or not, 
but if you want to test a 4 mon party with the rus,
import the everything data and it runs smooth :) 

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I considered writing automated tests for the issue?
- [x] If I have text, did I add placeholders for them in locales?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Comment by schmidtc1** [comment]
Likely relevant: #3501

File: src/ui/starter-select-ui-handler.ts [hunk 1]
```diff
@@ -875,29 +875,18 @@ export default class StarterSelectUiHandler extends MessageUiHandler {
...
-          generateSpecies();
-
-          for (let ps = 0; ps < c; ps++) {
-            if (this.pokerusSpecies[ps] === species) {
-              dupe = true;
-              break;
-            }
-          }
-        } while (dupe);
+        generateSpecies();
```
Comment (schmidtc1) [inline src/ui/starter-select-ui-handler.ts:889]: You can take the code out of `generateSpecies()` since you removed the do-while loop. Just place `randomSpeciesId`, `currentGenSpecies`, and `species` in `const` variables

**Review by schmidtc1 — CHANGES_REQUESTED** [review]

File: src/ui/starter-select-ui-handler.ts (hunk 1 above)
Comment (mKroogz) [reply src/ui/starter-select-ui-handler.ts:889]: You are so right!  I'm gonna do that right away.  :)

**Review by schmidtc1 — APPROVED** [review]
Lgtm, only concern would be overlap with #3501

**Comment by frutescens** [comment]
Hi! PR #3501 has been merged into the beta branch so hopefully your work will continue. :)

**Comment by damocleas** [comment]
Would like to ask this only be 5 pokemon, 9 feels a bit too high, and I feel this actually discourages using any pokemon that doesn't have it in a run..

**Comment by mKroogz** [comment]
> Hi! PR #3501 has been merged into the beta branch so hopefully your work will continue. :)

Yay I'm gonna keep working on it then :)

**Comment by frutescens** [comment]
Hi! The balance team will be adding an update that will increase the number of daily Pokerus starters to 5. I think that it might be good for there to be a check to make sure these 5 Pokerus starters are all from different generations. Would you be interested in changing your PR to focus on that? TY!!

**Comment by mKroogz** [comment]
> Hi! The balance team will be adding an update that will increase the number of daily Pokerus starters to 5. I think that it might be good for there to be a check to make sure these 5 Pokerus starters are all from different generations. Would you be interested in changing your PR to focus on that? TY!!

This sounds good to me I can do that

**Comment by DayKev** [comment]
> -Being able to have a pokerus mon for single gen challenges

Maybe challenges should have a way to modify which Pokémon have pokerus, and then if you're doing a monogen challenge it can check to see if there's a Pokémon of that gen with pokerus and if not add pokerus to one of the mons in that gen.
It would be part of a different PR though.

**Review by Tempo-anon — CHANGES_REQUESTED** [review]
More pokerus mons is good but 9 feels a bit too many. I do like the fact that they're more distributed throughout the generations though.

**Comment by MokaStitcher** [comment]
this code is quite out of date, starter select now calls an external function that gets the Pokerus species
https://github.com/pagefaultgames/pokerogue/blob/ddf97fd8f6129dfc1bab54da15f8c41655ed7380/src/ui/starter-select-ui-handler.ts#L927

FYI the number of Pokerus mons was just updated from 3 to 5 in #4268 

Can you update your PR to leave that number at 5 but still add your logic to `getPokerusSpecies` that makes sure they all come from different gens?

**Comment by Tempo-anon** [comment]
Going to close this since the PR is a bit out of date and the number of pokerus starters has been increased already.
